onnx/reference/ops/op_dequantize_linear.py

```diff
@@ -39,26 +39,43 @@ def reshape_input(
             raise ValueError("axis cannot be None.")
         if len(value.shape) == 0:
             return value
-        dims = [1] * len(shape)
-        try:
-            dims[axis] = value.size
-        except IndexError as e:
-            raise IndexError(
-                f"axis is out of boundary, axis={axis}, "
-                f"value.shape={value.shape}, shape={shape}."
-            ) from e
-        return value.reshape(tuple(dims))
+        if len(value.shape) == 1:
+            dims = [1] * len(shape)
+            try:
+                dims[axis] = value.size
+                return value.reshape(tuple(dims))
+            except IndexError as e:
+                raise IndexError(
+                    f"axis is out of boundary, axis={axis}, "
+                    f"value.shape={value.shape}, shape={shape}."
+                ) from e
+        if len(value.shape) >= 2:
+            if len(shape) != len(value.shape):
+                raise RuntimeError(
+                    "Input 2 must be a number, a vector or a tensor with the same rank as the input."
+                )
+            block_shape = np.array(shape) // np.array(value.shape)
+            if sum(block_shape != 1) != 1:
+                raise RuntimeError(
```

Comment:
Consider a ValueError?

Reply:
Yes, fixed in all applicable errors in op_dequantize_linear.py and in op_quantize_linear.py.